modules/images/webp-uploads/fallback.js

```diff
@@ -4,21 +4,32 @@ window.wpPerfLab = window.wpPerfLab || {};
 	window.wpPerfLab.webpUploadsFallbackWebpImages = function( media ) {
 		for ( var i = 0; i < media.length; i++ ) {
 			try {
-				if ( ! media[i].media_details.sources || ! media[i].media_details.sources['image/jpeg'] ) {
-					continue;
-				}
-
-				var ext = media[i].media_details.sources['image/jpeg'].file.match( /\.\w+$/i );
-				if ( ! ext || ! ext[0] ) {
-					continue;
-				}
+				var webp_sizes = window.wpPerfLabWebpSizes;
+				var size_keys = Object.keys( webp_sizes );
 
 				var images = document.querySelectorAll( 'img.wp-image-' + media[i].id );
 				for ( var j = 0; j < images.length; j++ ) {
-					images[j].src = images[j].src.replace( /\.webp$/i, ext[0] );
+					console.log( media[i] );
 					var srcset = images[j].getAttribute( 'srcset' );
 					if ( srcset ) {
-						images[j].setAttribute( 'srcset', srcset.replace( /\.webp(\s)/ig, ext[0] + '$1' ) );
+						// Update full image in srcset.
+						srcset = srcset.replace( images[j].src, media[i].media_details.sizes.full.source_url );
+						images[j].setAttribute( 'srcset', srcset );
+
+						// Update full image in src.
+						images[j].src = media[i].media_details.sizes.full.source_url;
+
+						// Update thumbnail images.
+						for ( var k = 0; k < size_keys.length; k++ ) {
+
+							if( ! media[i].media_details.sizes[size_keys[k]].sources['image/webp'] ) {
+								continue;
+							}
+
+							srcset = srcset.replace( media[i].media_details.sizes[size_keys[k]].sources['image/webp'].source_url, media[i].media_details.sizes[size_keys[k]].sources['image/jpeg'].source_url );
+							images[j].setAttribute( 'srcset', srcset );
```

Comment:
In addition to replacing the URL in the `srcset`, we also need to see if the WebP file for this size is used for the `src`, and if so, we need to update the `src` with the JPEG version. See my related above comments, `src` is not always the "full" size.

Reply:
Thank you for the review. Please check my other comments for review.